c++/src/Statistics.hh

```diff
@@ -41,49 +41,181 @@ namespace orc {
   };
 
 /**
+ * Internal Statistics Implementation
+ */
+
+  template <typename T>
+  class InternalStatisticsImpl {
+  private:
+    bool hasNull_;
```

Comment:
For the naming here, underscore is not consistent with the current code base. (Only current XXColumnStatistics classes use "_" in the beginning of members but I don't think they should be fixed)